### PR TITLE
chore(docs): restructuring of API documentation

### PR DESCRIPTION
**Pull request by butler54**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

- Users no longer need to update mkdocs to add a page.
- All pages have been moved to be in the same relative path for SEO
- Social previews will now be generated
- Descriptions for SEO
- Flavicons 

More work needs to be done.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by butler54** [comment]
@jpower432  / @degenaro a first pass review here would be helpful. All the descriptions (Where `BANANAS` has been put to have a key to replace) need to be updated.

Highlights:
1. Tree structure automatically creates the website
2. No need to update mkdocs.yaml to add a file. In 90% of locations this will happen automatically.

**Review by jpower432 — APPROVED** [review]
LGTM

Deployed locally and reviewed the changes. I don't see any issues with the changes made.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ repos:
       - id: check-yaml
         exclude: "tests/data/yaml/bad_simple.yaml"
       - id: no-commit-to-branch
-        args: [--branch, develop, --branch, main]
+        ar:gs: [--branch, develop, --branch, main]
```
[the remark this answers is not in the document]
Comment (butler54) [reply .pre-commit-config.yaml]: Thanks :)

**Comment by degenaro** [comment]
@butler54 I see you've already delivered. I pulled develop branch and doc serves works for me. Hopefully Reference -> API will be improved with future code restructuring.

**Pull request merged**